src/backend/BackendInterface.h

```diff
@@ -94,24 +94,22 @@ auto
 synchronous(F&& f)
 {
     boost::asio::io_context ctx;
-    boost::asio::strand<boost::asio::io_context::executor_type> strand(ctx.get_executor());
 
-    using R = typename boost::result_of<F(boost::asio::yield_context&)>::type;
+    using R = typename boost::result_of<F(boost::asio::yield_context)>::type;
     if constexpr (!std::is_same<R, void>::value)
     {
         R res;
-        boost::asio::spawn(
-            strand, [&f, &res, _ = boost::asio::make_work_guard(strand)](boost::asio::yield_context yield) {
-                res = f(yield);
-                ;
-            });
+        boost::asio::spawn(ctx, [&f, &res, _ = boost::asio::make_work_guard(ctx)](boost::asio::yield_context yield) {
```

Comment:
not sure i understand the code . why we need the extra parameter , but we don't use it inside?

Reply:
i think in this particular case we may not actually need this work guard - it was added for a test and i must have forgotten about it. thanks for noticing 👍 

as for the extra `;` - it's there to make clang-format format the body onto its own line instead of squashing everything into one unreadable line.. if i remove the work guard then maybe we can remove `;` too.